File: Linguagem de Programação I/Extras/Beecrowd/1222.py
import sys

for line in sys.stdin:
    n, l, c = map(int, line.split())
    words = input().split()
    lines = chars = pages = 0
    for word in words:
        if chars + len(word) > c:
            chars = 0
            lines += 1
            if lines == l:
                lines = 0
                pages += 1
        chars += len(word) + 1
    if chars or lines:
        pages += 1
    print(pages)

File: Linguagem de Programação I/Extras/Beecrowd_Extra/1222.cpp
#include <iostream>
#include <vector>
#include <string>

int main() {
    int n, l, c;
    while (std::cin >> n >> l >> c) {
        std::vector<std::string> words(n);
        for (int i = 0; i < n; i++) {
            std::cin >> words[i];
        }
        int lines = 0, chars = 0, pages = 0;
        for (int i = 0; i < n; i++) {
            int len = words[i].size();
            if (chars + len > c) {
                chars = 0;
                lines++;
                if (lines == l) {
                    lines = 0;
                    pages++;
                }
            }
            chars += len + 1;
        }
        if (chars || lines) {
            pages++;
        }
        std::cout << pages << std::endl;
    }
    return 0;
}